src/gui/macutilities.mm

```diff
@@ -85,16 +89,36 @@ void overrideDockClickHandler(bool (*dockClickHandler)(id, SEL, ...))
         }
     }
 
-    void displayNotification(const QString &title, const QString &message)
+    void askForNotificationPermission()
     {
         @autoreleasepool
         {
-            NSUserNotification *notification = [[NSUserNotification alloc] init];
-            notification.title = title.toNSString();
-            notification.informativeText = message.toNSString();
-            notification.soundName = NSUserNotificationDefaultSoundName;
+            [UNUserNotificationCenter.currentNotificationCenter requestAuthorizationWithOptions:
+                    (UNAuthorizationOptionAlert + UNAuthorizationOptionSound)
+                            completionHandler:^([[maybe_unused]] BOOL granted, NSError * _Nullable error)
+                            {
+                                if (error)
+                                {
+                                    LogMsg(QCoreApplication::translate("MacUtils", "Failed to grant permission for notifications. Error: \"%1\"").arg
```

Comment:
As mentioned in https://github.com/qbittorrent/qBittorrent/pull/23019#discussion_r2232791183, the log message would be wrong if `granted == false` since the user never grant permission in the first place.
Please consider changing the log message to something else.